src/app/login-button/login-button.component.ts

```diff
@@ -0,0 +1,27 @@
+// Adding back Alexander's work
+import { Component, OnInit } from '@angular/core';
+import {Router} from '@angular/router';
+import { AuthService } from '../core/auth.service';
+
+@Component({
+  selector: 'app-login-button',
+  templateUrl: './login-button.component.html',
+  styleUrls: ['./login-button.component.scss']
+})
+export class LoginButtonComponent implements OnInit {
+
+  constructor(public auth: AuthService, private router: Router) { }
+
+  ngOnInit(): void {
+    const loginButton = document.getElementById('login_button');
```

Comment:
I don't think this line even works. The whole point of using a framework like Angular is to prevent doing DOM manipulation like this.

Reply:
idk I copied what Alexander did (he put them in two separate components), and I just mashed them together